src/sys/socket/mod.rs

```diff
@@ -375,6 +375,32 @@ impl Ipv6MembershipRequest {
     }
 }
 
+cfg_if! {
+    if #[cfg(not(all(target_os = "linux",
+                     target_env = "gnu",
+                     any(target_arch = "mips",
+                         target_arch = "mips64",
+                         target_arch = "aarch64",
+                         target_arch = "arm",
+                         target_arch = "powerpc64"))))] {
+        /// Request for multicast socket's outgoing interface
+        ///
+        /// The value is a `libc::in_addr` structure, which is the most common.
+        /// `ip_mreqn` or `ip_mreq` structure are not supported at this point, one
+        /// reason is that if we use enum to represent all possible structures,
+        /// `#[repr(transparent)]` does not support enum.
+        #[repr(transparent)]
```

Comment:
Does the option work with IPv6?  If so you should use an enum type.  Or maybe define different Nix sockopts, one for each supported address type.